lite/kernels/nnadapter/engine.cc

```diff
@@ -329,49 +339,55 @@ bool Engine::Run() {
   for (size_t i = 0; i < input_count; i++) {
     input_dims[i] = input_vars_[i].value->dims().Vectorize();
   }
-  // Find the compiled device program according to the input dimensions
-  std::shared_ptr<Program> program = nullptr;
-  if (!programs_.count(input_dims)) {
-    // Rebuild the device program corresponding to the input dimensions if not
-    // found
-    std::vector<std::string> device_names;
-    for (auto* device : devices_) {
-      const char* name = nullptr;
-      NNAdapterDevice_getName_invoke(device, &name);
-      device_names.push_back(name);
+  // try to execute all programs.
+  for (auto program : programs_) {
+    int ret = program->Execute();
+    if (ret == NNADAPTER_INVALID_DIMENSIONS) {
+      VLOG(1)
+          << "Input shapes are invalid for the program, try the next program.";
```

Comment:
Input shapes are not supported by the program, try the next program.

Reply:
done